### PR TITLE
756 automated new business email

### PR DESCRIPTION
**Pull request by accua**
This PR establishes automated new business emails with a simple test. It also adds assertions to the `developer_mailer_test` file

link to issue: https://github.com/joemasilotti/railsdevs.com/issues/765

### CODE REVIEW

**Comment by accua** [comment]
Hey @joemasilotti, let me know if this looks like a good implementation to you! I left a TODO for the `first message` test in `developers_mailer_test.rb`.

I don't love the `assert_enqueued_email` assertion, but attempting to scope it to a mailer and method was giving me errors. Once we get this nailed down I'm happy to grab the other mailer ticket as well!

File: test/mailers/business_mailer_test.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    assert_enqueued_emails 1
+  end
+
+  def business
+    businesses(:one)
+  end
+
+  def recipient
+    business.user
+  end
```
Comment (joemasilotti) [inline test/mailers/business_mailer_test.rb:15]: This method isn't used, let's remove it.

File: test/mailers/business_mailer_test.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require "test_helper"
+
+class BusinessMailerTest < ActionMailer::TestCase
+  test "calling welcome enqueues the mailer" do
+    BusinessMailer.with(business:).welcome.deliver_later
+    assert_enqueued_emails 1
```
Comment (joemasilotti) [inline test/mailers/business_mailer_test.rb:6]:
```suggestion
    assert_emails 1 do
      BusinessMailer.with(business:).welcome.deliver_later
    end
```

File: app/views/business_mailer/welcome.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+<p>The platform is a bit different than traditional job boards. Instead of posting a job description, you <%= link_to "browse developer profiles", developers_url  %> to find a perfect match.</p>
+
+<p>When you see a good fit click the Hire button. Your message goes right to their email inbox.</p>
+
+<p><b>This approach has a ~60% response rate and can dramatically shorten the hiring process.</b> I've seen offers made in as little as 2 weeks.</p>
+
+<p>What are you looking for in your next hire? I'll send over some folks who could be a good fit.</p>
+
+<%= render "shared/signature" %>
```
Comment (joemasilotti) [inline app/views/business_mailer/welcome.html.erb:13]: Please remove the blank lines to match `app/views/developer_mailer/welcome.html.erb`.

File: test/mailers/developer_mailer_test.rb [hunk 4]
```diff
@@ -2,14 +2,19 @@
...
 
   test "welcome" do
-    Developers::WelcomeNotification.with(developer:).deliver(recipient)
+    assert_emails 1 do
+      Developers::WelcomeNotification.with(developer:).deliver(recipient)
+    end
   end
 
   test "first message" do
+    # TODO: Make an assertion here
```
Comment (joemasilotti) [inline test/mailers/developer_mailer_test.rb:17]: Let's remove this TODO and create a new issue in the repo.

File: test/mailers/business_mailer_test.rb (hunk 2 above)
Comment (joemasilotti) [reply test/mailers/business_mailer_test.rb:6]: Also, I think I'd rather create/send a notification to better match the developer mailer test.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Going through the flow I'm realizing we actually don't want a notification.

A notification can only display the title and link to a URL. Linking to `/developers` might be helpful, but there isn't enough context to give in just a title. (This also applies to the existing developer welcome flow.)

Instead, we have two options. After creating a business (or developer) profile:

1. Don't use a notification - only use an email.
2. Redirect to a success page with similar copy to the email.

I'd rather see the second but it will require more code changes.

So, for this PR, let's wrap the sending of the email in a feature flag. Add a new one to `Feature.rb` and only send the notification + email if it is enabled.

Then, once this is merged, we can add the new routes for `/businesses/welcome` and `/developers/welcome` that render designed copy like the emails.

**Comment by accua** [comment]
Hey @joemasilotti,

Just writing to let you know that I plan to get this wrapped up in the near future. I'm currently on vacation, but brought my laptop and am juggling this PR and interview code challenges

**Comment by joemasilotti** [comment]
Sounds great! No rush on it and thanks for the update. Let me know if you need any help.

**Comment by joemasilotti** [comment]
Hey again @accua, two week check in. Do you still have appetite to finish this PR?

If not, no worries. I can take it over the finish line.

**Comment by accua** [comment]
Hey Joe! Happy to bring this across the finish line. How does complete by midweek sound?On Feb 26, 2023, at 12:58 PM, Joe Masilotti ***@***.***> wrote:﻿
Hey again @accua, two week check in. Do you still have appetite to finish this PR?
If not, no worries. I can take it over the finish line.

**Comment by joemasilotti** [comment]
Sounds great to me, thanks! I'm here if you need anything.

File: test/mailers/business_mailer_test.rb (hunk 1 above)
Comment (accua) [reply test/mailers/business_mailer_test.rb:15]: When changing the assertion to match the `developer_mailer_test`, this method ended up being used. 👍

File: test/mailers/developer_mailer_test.rb (hunk 4 above)
Comment (accua) [reply test/mailers/developer_mailer_test.rb:17]: Created as a bug

**Comment by accua** [comment]
Hey @joemasilotti! I had some spare time this evening and went ahead and made the changes listed above. Let me know your thoughts 😄

**Review by joemasilotti — APPROVED** [review]

**Pull request merged**